src/Input.js

```diff
@@ -120,7 +121,8 @@ class Input extends Component {
         inputType = type || 'text';
     }
     let labelClasses = {
-      active: this.state.value || this.isSelect()
+      active: this.state.value || this.isSelect() || this.props.value ||
```

Comment:
Let's wrap this into a function i.e. `isActive()`, The problem here is not that `props.value` is not checked (https://github.com/react-materialize/react-materialize/blob/master/src/Input.js#L13), but rather in js `0` is `false`. Probably a better way to describe it is:
```js
isActive: () -> {
    this.state.value !== '' || this.isSelect()
}
```